### PR TITLE
Add config controller stub

### PR DESCRIPTION
**Pull request by kopiczko**
Towards https://github.com/giantswarm/giantswarm/issues/15490

### CODE REVIEW

File: cmd/generate/runner.go [hunk 1]
```diff
@@ -106,7 +106,7 @@ func (r *runner) run(ctx context.Context, cmd *cobra.Command, args []string) err
 	gen, err := generator.New(&generator.Config{
 		Fs:               store,
 		DecryptTraverser: decryptTraverser,
-		ProjectVersion:   project.AppControlPlaneVersion(),
+		ProjectVersion:   project.Version(),
```
Comment (kopiczko) [inline cmd/generate/runner.go:109]: I changed that only to simplify the code. I don't think that it is important that the generated version is `0.0.0` when using CLI. I'm not even sure if it should be `0.0.0`.

File: service/controller/config.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
+package controller
```
Comment (kopiczko) [inline service/controller/config.go:1]: All below is some copy+paste + some adjustments with stub `configuration` resource.

File: service/controller/config.go [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+			Logger:    config.Logger,
+			NewRuntimeObjectFunc: func() runtime.Object {
+				return new(v1alpha1.Config)
+			},
+			Resources: resources,
+			Selector:  label.VersionSelector(config.UniqueConfig),
+
+			// Name is used to compute finalizer names. This here results in something
+			// like operatorkit.giantswarm.io/config-controller-config-controller.
+			Name: project.Name() + "-config-controller",
```
Comment (kopiczko) [inline service/controller/config.go:55]: Naming...

File: service/controller/handler/configuration/create.go [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package configuration
+
+import (
+	"context"
+)
+
+func (h *Handler) EnsureCreated(ctx context.Context, obj interface{}) error {
+	panic("not implemented")
```
Comment (kopiczko) [inline service/controller/handler/configuration/create.go:8]: To be implemented

File: go.mod [hunk 5]
```diff
@@ -8,7 +8,7 @@ require (
 	github.com/Masterminds/sprig v2.22.0+incompatible
 	github.com/fatih/color v1.9.0 // indirect
 	github.com/ghodss/yaml v1.0.0
-	github.com/giantswarm/apiextensions/v3 v3.14.1
+	github.com/giantswarm/apiextensions/v3 v3.16.2-0.20210202133116-302f4a6aacb5
```
Comment (kubasobon) [inline go.mod:11]: Is there a saner release of `apiextensions` that can be used?

Comment (kopiczko) [reply go.mod:11]: Not yet. My other PR to apiextensions is hanging.

Comment (kubasobon) [reply go.mod:11]: Excellent

File: service/controller/handler/configuration/handler.go [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+import (
+	"github.com/giantswarm/k8sclient/v5/pkg/k8sclient"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	vaultapi "github.com/hashicorp/vault/api"
+
+	"github.com/giantswarm/config-controller/pkg/k8sresource"
+)
+
```
Comment (kopiczko) [inline service/controller/handler/configuration/handler.go:11]: Only wiring here

**Review by kubasobon — APPROVED** [review]

**Pull request merged**